modules/collect/src/main/java/com/opengamma/strata/collect/array/BoxedDoubleArray.java

```diff
@@ -0,0 +1,224 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * 
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.collect.array;
+
+import java.io.Serializable;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.stream.DoubleStream;
+import java.util.stream.Stream;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanBuilder;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaBean;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyStyle;
+import org.joda.beans.impl.BasicImmutableBeanBuilder;
+import org.joda.beans.impl.BasicMetaBean;
+import org.joda.beans.impl.BasicMetaProperty;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.DoubleArrayMath;
+
+/**
+ * An immutable array of {@code double} values, boxed to {@code Double}.
+ * <p>
+ * This is used to wrap a {@code double[]} to meet the API of {@link Array}.
+ * An instance is created using {@link DoubleArray#boxed()}.
+ */
+final class BoxedDoubleArray
```

Comment:
I'm not sure about that. Having "scenario" in the name communicates the fact that it contains values for multiple scenarios. There is logic applied to those values to extract the single value when the user calls the single-scenario variants of `CalculationRunner.calculate*`. If we just use a general array type then users might use them for general arrays of values which might produce surprising behaviour.